File: rb/features.rb
# UToLinkShortener SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module UToLinkShortenerFeatures
  def self.make_feature(name)
    case name
    when "base"
      UToLinkShortenerBaseFeature.new
    when "test"
      UToLinkShortenerTestFeature.new
    else
      UToLinkShortenerBaseFeature.new
    end
  end
end

File: py/features.py
# UToLinkShortener SDK feature factory

from feature.base_feature import UToLinkShortenerBaseFeature
from feature.test_feature import UToLinkShortenerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UToLinkShortenerBaseFeature(),
        "test": lambda: UToLinkShortenerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
